src/ray/raylet/worker.h

```diff
@@ -219,6 +224,13 @@ class Worker : public WorkerInterface {
   RayTask &GetAssignedTask() { return assigned_task_; };
 
   void SetAssignedTask(const RayTask &assigned_task) {
+    const auto &task_spec = assigned_task.GetTaskSpecification();
+    SetJobId(task_spec.JobId());
+    SetBundleId(task_spec.PlacementGroupBundleId());
+    SetOwnerAddress(task_spec.CallerAddress());
+    AssignTaskId(task_spec.TaskId());
+    SetIsGpu(task_spec.GetRequiredResources().Get(scheduling::ResourceID::GPU()) > 0);
+    SetIsActorWorker(task_spec.IsActorCreationTask() || task_spec.IsActorTask());
```

Comment:
It should never be `ActorTask` given actor task directly goes to the caller instead of going through raylet.

Reply:
removed